individual/models.py

```diff
@@ -13,12 +15,50 @@ class Individual(HistoryModel):
     #TODO WHY the HistoryModel json_ext was not enough
     json_ext = models.JSONField(db_column="Json_ext", blank=True, default=dict)
 
+    village = models.ForeignKey(
+        Location,
+        models.DO_NOTHING,
+        blank=True,
+        null=True
+    )
+
     def __str__(self):
         return f'{self.first_name} {self.last_name}'
 
     class Meta:
         managed = True
 
+    @classmethod
+    def get_queryset(cls, queryset, user):
+        if queryset is None:
+            queryset = cls.objects.all()
+
+        if not settings.ROW_SECURITY:
+            return queryset
+
+        if user.is_anonymous:
+            return queryset.filter(id=-1)
+
+        if not user.is_imis_admin:
+            user_districts_match_individual = LocationManager().build_user_location_filter_query(
+                user._u,
+                prefix='village__parent__parent',
+                loc_types=['D']
```

Comment:
this will work fine but we might change it to support all locations to just prefix='location and no loc_types'
```suggestion
```